Домашнее задание № 13

aegorin opened the pull request.

Commit:
Домашнее задание № 13

vlad-iv commented on hw13-acl/src/main/java/ru/otus/hw/security/UserDetailsServiceJpa.java:
```diff
@@ -0,0 +1,48 @@
+
+@Service
+@RequiredArgsConstructor
+public class UserDetailsServiceJpa implements UserDetailsService {
```
Можно назвать JpaUserDetailsService

vlad-iv commented on hw13-acl/src/main/java/ru/otus/hw/services/BookServiceImpl.java:
```diff
@@ -0,0 +1,69 @@
+
+    @Override
+    @Transactional(readOnly = true)
+    @PreAuthorize("hasRole('USER')")
```
Предлагаю выбрать единый подход с доступом:
пример: все могут читать данные пользователи дополнительное еще создавать (и редактировать + удалять, но только свои данные - такое можно не делать) , а менеджеры (админы) удалять и редактировать любые 

и задать их в SecurityConfiguration
и только если необходимо что-то дополнительное то задавать уже в сервисах

Commit:
Переименовать класс

Commit:
Изменить правила предоставления прав пользователям

Пользователи группы USERS могут читать все записи.

Пользователи группы CUSTOMERS могут читать все записи
и могут добавлять в базу новые книги. А также редактировать
и удалять книги если пользователь является владельцем книги.

Пользователи группы MANAGERS могут читать все записи,
добавлять новые книги, редактировать/удалять любые книги.

vlad-iv commented on hw13-acl/src/main/java/ru/otus/hw/models/Book.java:
```diff
@@ -0,0 +1,46 @@
+
+    @ManyToOne(fetch = FetchType.LAZY)
+    @JoinColumn(name = "owner_id")
+    private User owner;
```
Можно применять подход с ACL
